From 41c3b5f18febda870940ae1b89c9b4437df4414a Mon Sep 17 00:00:00 2001
From: Juan Quintela <quintela@redhat.com>
Date: Wed, 19 Oct 2011 23:47:45 +0200
Subject: [PATCH 10/19] migration: set error if select return one error

RH-Author: Juan Quintela <quintela@redhat.com>
Message-id: <7e825fe72962daba9a2bd5c23d807c759ecb7170.1319066771.git.quintela@redhat.com>
Patchwork-id: 34437
O-Subject: [PATCH qemu-kvm RHEL-6.2 10/16] migration: set error if select return one error
Bugzilla: 669581
RH-Acked-by: Amit Shah <amit.shah@redhat.com>
RH-Acked-by: Markus Armbruster <armbru@redhat.com>
RH-Acked-by: Orit Wasserman <owasserm@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>

Signed-off-by: Juan Quintela <quintela@redhat.com>
Reviewed-by: Anthony Liguori <aliguori@us.ibm.com>
---
 migration.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 migration.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/migration.c b/migration.c
index dd65bf7..e35cb41 100644
--- a/migration.c
+++ b/migration.c
@@ -474,6 +474,10 @@ void migrate_fd_wait_for_unfreeze(void *opaque)
 
         ret = select(s->fd + 1, NULL, &wfds, NULL, NULL);
     } while (ret == -1 && (s->get_error(s)) == EINTR);
+
+    if (ret == -1) {
+        qemu_file_set_error(s->file);
+    }
 }
 
 int migrate_fd_close(void *opaque)
-- 
1.7.4.4

